### ISSUE TITLE
AMD nested virtualization?

### ISSUE THREAD

**Issue by SidShetye** [MicrosoftDocs/Virtualization-Documentation#1276]
We have CI/CD VMs (Jenkins) running on Hyper-V that tests Docker based builds. Docker in turn requires Hyper-V, so we're unable to run any of those Jenkins' VMs on our AMD Ryzen based servers.

How do we resolve this? AMD support seems to point the finger at Microsoft Hyper-V.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 9a13cf32-4be1-94d9-b764-0faf79372065
* Version Independent ID: 6787036e-5a30-bd17-e87a-54f61b0ba47f
* Content: [Nested Virtualization](https://docs.microsoft.com/en-us/virtualization/hyper-v-on-windows/user-guide/nested-virtualization?redirectedfrom=MSDN)
* Content Source: [virtualization/hyper-v-on-windows/user-guide/nested-virtualization.md](https://github.com/Microsoft/Virtualization-Documentation/blob/live/virtualization/hyper-v-on-windows/user-guide/nested-virtualization.md)
* Service: **windows-10-hyperv**
* Product: **windows-10-hyperv**
* GitHub Login: @johncslack

**Comment by heymrdjcw** [comment]
I too need this answer. We won’t continue to support Windows 10 if Microsoft won’t support the processor company that’s actually delivering value at this point.

**Comment by Larswa** [comment]
What processor are you running?  I just built a new developer machine with an AMD chipset and an AMD ryzen 9 3900x. And hyper-v virtualization works fine with no issues.

You have to enable virtualization in the UEFI/BIOS BEFORE you install hyper-v but apart from that there were no issues. As stated, I am running machines on HyperV, and before that Virtual box workloads without any issues. It just works. Also running WSL, Minikube, Kind and other such dev related tools ... no issues there. Works just as fine as my old i7.

**Comment by heymrdjcw** [comment]
Please keep in mind this discussion is about NESTED virtualization, a feature of Windows 10, Server 2016, and Server 2019. Attempt to run a VM in Hyper-V that also has WSL or Hyper-V enabled. This feature is extremely important for VDI farms, development servers, production private cloud servers, and compact labs.

**Comment by Larswa** [comment]
Yup. I'm running nested virtualization, doing windows docker hyperv containers in a hyperv guest. Works fine. What exactly is the problem you are seeing?

**Comment by SidShetye** [comment]
Can you share your processor and OS specs? The title indicates AMD processors ...

On Nov 28, 2019, at 4:27 PM, Larswa <notifications@github.com> wrote:

﻿

Yup. I'm running nested virtualization, doing windows docker hyperv containers in a hyperv guest. Works fine. What exactly is the problem you are seeing?

**Comment by Larswa** [comment]
Happy to..  Here are screenshots from taskman showing that windows are seeing virtualization as enabled, and a screenshot of cpu-z.  I can highly recommend getting 12 core in a machine ... :-D  Parallel  workloads (like building in parallel) doesnt slow the machine down noticeably. 

![Taskman 2019-11-28 22_32_31-Window](https://user-images.githubusercontent.com/1431240/69832636-c66cbf80-122f-11ea-922d-a22e902b3335.png)
![cpu-z 2019-11-28 22_33_08-Window](https://user-images.githubusercontent.com/1431240/69832642-c8cf1980-122f-11ea-9af7-5975401a1fb5.png)

**Comment by Larswa** [comment]
But @SidShetye - what problem are you getting when installing/enabling hyperv or starting a hyperv machine on your AMD machines?

**Comment by heymrdjcw** [comment]
@Larswa , the issue is the typical no start condition as shown by hundreds of posts across the internet of people trying. Your experience runs opposite of that, as well as Microsoft's own statements. Can you please attempt to start a full Windows VM on your Windows Hyper-V VM (10, 2016, or 2019) running on your Hyper-V AMD system (10, 2016, or 2019). That should be a 3rd level Windows system running. Your current screenshots don't show you running a VM on a Hyper-V server VM that is on a Hyper-V physical server.

**Comment by SidShetye** [comment]
I’m away from our lab for thanksgiving to share the exact error but the issue we faced is hyper-v cannot be enabled inside the VM to have 2nd levels of VM inside the first levels of VM. On all generation 1 8C/16T Ryzens

On Nov 28, 2019, at 4:38 PM, Larswa <notifications@github.com> wrote:

﻿

Happy to.. Here are screenshots from taskman showing that windows are seeing virtualization as enabled, and a screenshot of cpu-z. I can highly recommend getting 12 core in a machine ... :-D Parallel workloads (like building in parallel) doesnt slow the machine down noticeably.

[Taskman 2019-11-28 22_32_31-Window]<https://user-images.githubusercontent.com/1431240/69832636-c66cbf80-122f-11ea-922d-a22e902b3335.png>
[cpu-z 2019-11-28 22_33_08-Window]<https://user-images.githubusercontent.com/1431240/69832642-c8cf1980-122f-11ea-9af7-5975401a1fb5.png>

**Comment by Larswa** [comment]
No oh god ... youre right. I am getting the same error. How annoying. I was running docker containers in my guest, but I see now that they were NOT running in Hyperv mode, but in isolation. So I stand corrected. Well .. lets hope MS get this sorted. Untill then KVM and allegedly Virtualbox supports nested virtualization but I havent tested them. I can do with running containers on guests and not go any deeper. Sorry for the confusion guys. :-)

**Comment by ImreKovats** [comment]
This is indeed a very good question by the Sid.
Seems to be a product decision (or lack of development time) on Microsoft's part, rather then a technical limitation of AMD virtualization extensions.

Honestly, one might ponder to just switch to vmware as it does support nested virtualization workloads on both AMD and Intel CPUs.

**Comment by plukawski** [comment]
That issue with nested virtualization is being discussed highly since first Ryzen launched in 2017. I think the best way is to upvote that feature massively that on MS Feedback HUB and on user voice portal. Without that I don't think so if MS will put more resources to address that :(
I have already upvoted that feature in those places. I encourage you all to also do the same, so that strange limitation is lifted in a near future.
Here is the link to User Voice portal about that feature: https://windowsserver.uservoice.com/forums/295047-general-feedback/suggestions/31734808-nested-virtualization-for-amd-epyc-and-ryzen

**Comment by JouMxyzptlk** [comment]
I would love to have that feature, at least in Server 2019 / Windows 10 1809+. I still have to use my old i7-4960x for that. The AMD threads on that: https://community.amd.com/thread/247222 and https://community.amd.com/thread/224825

**Comment by Dresel** [comment]
Update from the user voice [portal](https://windowsserver.uservoice.com/forums/295047-general-feedback/suggestions/31734808-nested-virtualization-for-amd-epyc-and-ryzen):

> Hello Hyper-V fans! I was recently alerted to your feedback on this topic and I hear your request loud and clear! I run the engineering team that is responsible for the Hyper-V hypervisor. I cannot commit to a specific Windows release that will support nested virtualization on an AMD CPU at the moment, but I recommend monitoring Windows insider builds for previews of this support as we push the changes into the hypervisor this year.

**Comment by danigt91** [comment]
Any news? Some dates? I need this feature so hard.
i.e. Not being able to have a docker on a VM is a big stopper.

**Comment by idprism** [comment]
I just found out that I can't develop on a vm on HyperV because of this, and it is quite frustrating. I may have to relegate windows to go to qemu/kvm and use windows only as a VM if the Microsoft devs can't get their act together.

**Comment by LiuChangFreeman** [comment]
> Update from the user voice [portal](https://windowsserver.uservoice.com/forums/295047-general-feedback/suggestions/31734808-nested-virtualization-for-amd-epyc-and-ryzen):
> 
> > Hello Hyper-V fans! I was recently alerted to your feedback on this topic and I hear your request loud and clear! I run the engineering team that is responsible for the Hyper-V hypervisor. I cannot commit to a specific Windows release that will support nested virtualization on an AMD CPU at the moment, but I recommend monitoring Windows insider builds for previews of this support as we push the changes into the hypervisor this year.

Hope this feature will come soon.

**Comment by lvasiliou** [comment]
Its probably easier to virtualise windows in linux at this stage as Microsoft dont want to support nested virtualisation on AMD hardware

**Comment by Dresel** [comment]
> 
> 
> Its probably easier to virtualise windows in linux at this stage as Microsoft dont want to support nested virtualisation on AMD hardware

That's what I'm doing now, but without Hyper-V, Docker and friends it's only half the fun.

**Comment by lvasiliou** [comment]
> 
> 
> > Its probably easier to virtualise windows in linux at this stage as Microsoft dont want to support nested virtualisation on AMD hardware
> 
> That's what I'm doing now, but without Hyper-V, Docker and friends it's only half the fun.

If you install May 2020 WIndows 10 Update you can run docker on wsl2 without Hyper-V.

**Comment by Dresel** [comment]
> 
> 
> > > Its probably easier to virtualise windows in linux at this stage as Microsoft dont want to support nested virtualisation on AMD hardware
> > 
> > 
> > That's what I'm doing now, but without Hyper-V, Docker and friends it's only half the fun.
> 
> If you install May 2020 WIndows 10 Update you can run docker on wsl2 without Hyper-V.

That's good to know - still need Hyper-V for other use cases.

**Comment by wbsmolen** [comment]
> > > Its probably easier to virtualise windows in linux at this stage as Microsoft dont want to support nested virtualisation on AMD hardware
> > 
> > 
> > That's what I'm doing now, but without Hyper-V, Docker and friends it's only half the fun.
> 
> If you install May 2020 WIndows 10 Update you can run docker on wsl2 without Hyper-V.

wsl2 can't be enabled without turning on hyper-v bits.

i run a kvm host with a win10 guest on an x570 taichi with a ryzen 3900x. i've been struggling for months to get this scenario working and have given up. the problem isn't unique to hyper-v hosts. 

also hoping to flag attention the kvm issues. through trial/error, I learned the +vmx qemu cpu boot arg is not supported on amd platforms. this seems to [functionally] be the missing piece on amd-based kvm hosts. I get BSODs with "hypervisor_error" if I boot with any hyper-v bits enabled (wsl, credential guard, hyper-v, etc).

**Comment by stuaxo** [comment]
@wbsmolen do you know if there is a kvm bug report for this ?

**Comment by wbsmolen** [comment]
@stuaxo no, but im not convinced its a kvm bug either.

**Comment by stuaxo** [comment]
It's probably worth making one, just in case it is - and if it isn't someone there might have an idea about where a fix is needed.

**Comment by lukts30** [comment]
`kvm_amd` has support for nesting linux with kvm guests. 
`kvm_intel` can also host hyper-v enabled windows machines. [[2017]Nested Virtualization: Hyper-V on KVM by Ladi Prosek](https://www.youtube.com/watch?v=zQSkKgpj9jA)

You can not expect to run hyper-v on kvm when native nesting (hyper-v on hyper-v) is not even possible. 
https://bugzilla.kernel.org/show_bug.cgi?id=203477

**Comment by wbsmolen** [comment]
>You can not expect to run hyper-v on kvm when native nesting (hyper-v on hyper-v) is not even possible. 

I've done exactly this on intel machines. Does hyper-v function normally in the guest? No. But windows boots and [for reasons] that's all I'm looking for in my amd environment

**Comment by nikopen** [comment]
> 
> 
> > It's your CPU from AMD? It's well known nested Hyper-v virtualization doesn't work on AMD cpus.
> 
> Been meaning to backlink [this uservoice](https://windowsserver.uservoice.com/forums/295047-general-feedback/suggestions/31734808-nested-virtualization-for-amd-epyc-and-ryzen) for a while. Encourage everyone to upvote. [Discourage everyone from adding a vacuous quip like the sorry state of the comment section over there.]
> 
> ![image](https://user-images.githubusercontent.com/7049700/82678612-be01aa00-9bfe-11ea-99b1-4ec4494aeaf9.png)

bump

**Comment by plukawski** [comment]
Until now I can confirm that nesting on AMD works without issue on VMWare ESXi 6.7 U1 - I am able to enable hyper-v in windows guest and then use docker desktop in that VM without any issues. 
So it seems that for now VMWare ESXi is the only viable option for nesting on AMD platform.

**Comment by wbsmolen** [comment]
@plukawski thanks for sharing, helpful information. if you startup a hyper-v vm in the guest vm, do you get any errors? asking out of curiosity. I could also bootup a hyper-v-enabled guest vm on a kvm host, but running hyper-v vms on the guest errored out. 

wsl2 seemingly was unaffected, which is interesting.

**Comment by asimonf** [comment]
3 years on and this hasn't been solved. It's so weird to be honest...

**Comment by plukawski** [comment]
@wbsmolen as to the VMWare ESXi the hyper-v in the guest vm is working fully without any issue for me (no crashes, no freezes, etc.). Like I said - docker desktop launched without any problem and it is using Linux Virtual machine in hyper-v in the backround. So ESXi is fully stable and operational when nesting on AMD.
As to your code 43, this is a hardcoded limitation in nVidia driver if I recall correctly - only Tesla cards can work that way properly, because only Tesla allow virtualization of GPU.

**Comment by lvasiliou** [comment]
Trying to understand, are you saying you cant run docker on wsl2 with AMD on Windows 10?

**Comment by plukawski** [comment]
@lvasiliou I don't know how you came into that conclusion. This topic is about nested virtualization (virtual machine inside virtual machine). You can launch wsl2 with AMD on bare metal without any issue. 
You just cannot do that in a virtual machine that runs in HYPER-V hosted on AMD platform.

**Comment by wbsmolen** [comment]
> As to your code 43, this is a hardcoded limitation in nVidia driver if I recall correctly - only Tesla cards can work that way properly, because only Tesla allow virtualization of GPU.

@plukawski oddly enough I have no issues passing the gtx1080 on the kvm host when I don't enable hyper-v, which is where im currently at. can also pass a 5700xt without issue but that's loosely related here.

**Comment by heymrdjcw** [comment]
> > As to your code 43, this is a hardcoded limitation in nVidia driver if I recall correctly - only Tesla cards can work that way properly, because only Tesla allow virtualization of GPU.
> 
> @plukawski oddly enough I have no issues passing the gtx1080 on the kvm host when I don't enable hyper-v, which is where im currently at. maybe it's kvm hiding itself from the guest? im not explicitly passing that param in the vm's xml config.
> 
> can also pass a 5700xt without issue but that's loosely related here.

KVM has the ability to hide the hypervisor layer from the virtualized OS, which fools the driver. Hyper-V doesn't have any ability to do this. For awhile, the community was patching nVidia drivers to delete this check, but I don't think that's worked for awhile either. https://mathiashueber.com/fighting-error-43-nvidia-gpu-virtual-machine/

**Comment by wbsmolen** [comment]
@heymrdjcw I'm aware, but as I said I'm not explicitly passing the kvm hidden state to the vm

**Comment by MatthiasTams** [comment]
I just tried wsl2 inside a Windows 10 Pro VM running as a guest on my Hyper-V Host (Windows 10 Pro) with an AMD Ryzen 7 2700X. Seems not to work, because wsl2 needs the hypervisor. So sad, even on Windows 10 2004.

**Comment by plukawski** [comment]
@MatthiasTams could you try to forcibly enable hyper-v in your nested VM using the following command:
```
Enable-WindowsOptionalFeature -Online -FeatureName Microsoft-Hyper-V -All
```

and then try the wsl2 again? Maybe for full virtual machine inside vm this will not work but for wsl2 it may work. I think this is nice to try.

**Comment by MatthiasTams** [comment]
> @MatthiasTams could you try to forcibly enable hyper-v in your nested VM using the following command...

Thank you for the suggestion. I just tried it. Unfortunately the same error appears as before. The message asks me to activate the "Virtual Machine Platform" (which I already did) and quits.

Btw: If I try to enable the nested virtualization with
  Set-VMProcessor -VMName <VMName> -ExposeVirtualizationExtensions $true
the VM refuses to start with the message that nested virtualization is not supported on this platform.

Sad. Still no support for nested VT on my AMD Ryzen 7 2700X and sad that wsl2 needs that.
Anyway, wsl2 is very useful on my host, even if it is unusable inside guest VMs on this host...

Now, I have another elderly workstation here with some "slow" Xeon 4-core, I will try it on that machine :-)

**Comment by lvasiliou** [comment]
The situation is beyond ridiculous at this stage, can we get an official response about Microsoft's position on this besides added to backlog and working on it?
I would suspected Microsoft would love to run EPYC on Azure as well at this stage, unless they are doing it with linux there as well

**Comment by lvasiliou** [comment]
So I spent ages trying to figure out nested virtualisation on KVM with Windows 10 2004 as a guest. Very hard to trick it to go past turn windows features on or off as it insists i dont have SLAT. Also as much as i tried to hide the fact its a VM, Task Manager kept reporting Virtual Machine: Yes. I tried all the KVM hiding tricks, reporting an Intel CPU etc. It seems the windows team knows people have been doing this. AWS has EPYC servers now by the way because they recognise $ per Watt.

**Comment by fre4ki** [comment]
Hey guys,

i have also an issue with nested, but with VMware ESXi 7 -> Ubuntu 20.04 LTS VM -> Qemu 4.2/5

System: Ryzen 9 - 3950X

You can watch this here:

https://github.com/GNS3/gns3-server/issues/1774

Since i changed the ignore_msrs parameter I get much messages in the console screen, but the VMs are starting. Maybe this can help someone..

**Comment by wbsmolen** [comment]
so great news -- win10 build 19640 (in insider's ring) has hyper-v support for nested virtualization. does not work on kvm hosts.

**Comment by lvasiliou** [comment]
That's great news. Is it available yet or just internal testing?

**Comment by wbsmolen** [comment]
Available to the public via the insider's fast ring :)

**Comment by plukawski** [comment]
Can someone tell me why it is not mentioned here? https://blogs.windows.com/windowsexperience/2020/06/03/announcing-windows-10-insider-preview-build-19640/

Such an important feature just not mentioned at all... :(

**Comment by wbsmolen** [comment]
I honestly don't know. I'm not privy to anything you do not know -- I can just vouch for personal experience. Quite possibly there are more comprehensive docs in the works.

**Comment by lvasiliou** [comment]
Great but I am staying away until the gaming issues are fixed :D

**Comment by dstaley** [comment]
@wbsmolen Do you know if that update enables the [Windows 10X emulator](https://docs.microsoft.com/en-us/dual-screen/windows/get-dev-tools) on AMD CPUs?

**Comment by wbsmolen** [comment]
@dstaley haven't tested it, can't vouch for that. I got a win10 nested vm to work on a hyper-v host running a 3900x with out-of-box hyper-v settings.

**Comment by lvasiliou** [comment]
@dstaley, if you look at the Windows 10X emulator on the Microsoft store, under system requirements it now says you must install windows 10 19500.0 or higher. Previously it used to say it doesnt work on AMD. I will imagex my PC just in case I want to get back to 1909 and I will give 2004 another ago.

**Comment by Larswa** [comment]
Hey @lvasiliou ... I just got my hopes up that they had committed some code in the fast ring for this, BUT the Microsoft Documentation page still states that AMD is not supported. And it was last updated 2nd June.   Saaaad Panda. 

https://docs.microsoft.com/en-us/dual-screen/windows/get-dev-tools

Still looking forward to you results on a Fast Ring build though. Ever the optimist, eh?

**Comment by lvasiliou** [comment]
@Larswa. I doubt production documentation will talk about Insider Preview features. I could be wrong. 
My full system backup is at 21%....

**Comment by lvasiliou** [comment]
Confirmed Windows 10X Emulator working on latest Windows Insider with AMD Ryzen Threadripper 3970X
![image](https://user-images.githubusercontent.com/8229290/83981049-44fa9600-a912-11ea-8003-5b311a534d88.png)

**Comment by dstaley** [comment]
Now the question is whether we have to wait until 20H2 to get this, or if it'll make its way into a 2004 build.

**Comment by JouMxyzptlk** [comment]
> 
> 
> so great news -- win10 build 19640 (in insider's fast ring) has hyper-v support for nested virtualization on amd systems. 

Confirmed! Update-VMVersion does not work yet, and you have to create the machine with Powershell using:
New-VM -Name "S2019 Nested Test" -Version 9.3 -Generation 2
If you don't supply a version a 9.0 machine is created. Currently installing Server 2019 to run part two of the test, activating Hyper-V in there.

**Comment by lvasiliou** [comment]
Deleted my VM, created a new one with ver 9.3 and attached my vhdx however i still get your CPU does not support virtualisation when trying to add hyper-v

**Comment by plukawski** [comment]
I confirm that this is indeed working:) I was able to create a nested VM with Windows 10 1909 PRO as a guest, activate hyper-V and then install docker desktop with linux containers. Then I checked if linux containers work and they worked :)

But you have to do what @JouMxyzptlk said in order to create a nested VM. 
So having the build 19640 on host allows you to create VMs with nested virtualization working on AMD.

Tested this on Ryzen 7 3700X.

**Comment by JouMxyzptlk** [comment]
> 
> 
> > so great news -- win10 build 19640 (in insider's fast ring) has hyper-v support for nested virtualization on amd systems.
> 
> Confirmed! Update-VMVersion does not work yet, and you have to create the machine with Powershell using:
> New-VM -Name "S2019 Nested Test" -Version 9.3 -Generation 2
> If you don't supply a version a 9.0 machine is created. Currently installing Server 2019 to run part two of the test, activating Hyper-V in there.

Can't edit any more, here the confirmation: It works! All relevant stuff is in that screenshot. CPU Ryzen 3900x, 64 GB ECC RAM.
![Win10-AMD3900x-nested-virtualization-works!](https://user-images.githubusercontent.com/10100281/83981846-f1904400-a921-11ea-9186-bb7f043bf54b.PNG)

**Comment by lvasiliou** [comment]
Works on the Threadripper 3970X as well. I forgot to configure the vprocessor and thats why it didnt let me add it via turn windows features on or off

**Comment by Kurnn** [comment]
I Have a new ASUS TUF B450-PLUS GAMING motherboard and Ryzen 5 3600Ghz CPU.
Win 10, home, 64 bit.

BIOS says 
"HI dude, you have secure boot, lots of TPM keys and I can overclock to you hell if you want"
"Oh yeah, CSM is on enabled, which means you can boot on both Legacy boot and UEFI! Isnt that great?

![Screenshot (4)](https://user-images.githubusercontent.com/45059948/83984746-e1369400-a936-11ea-861b-61106db05e87.png)


Windows says:
- Nope, If you Virtualize AMD CPU, I'll CRASH
- No secure boot loaded for you!
- TPM looks great, please clear your TPM (after 4 times, I stopped trying)

![Screenshot (22)](https://user-images.githubusercontent.com/45059948/83984769-f6abbe00-a936-11ea-9c11-312e5406444a.png)


- Windows further claim I have very little hardware protection. Not even ...well you check the pics.

![Screenshot (28)](https://user-images.githubusercontent.com/45059948/83984831-3ffc0d80-a937-11ea-86b7-a44e65dc0564.png)


I Hope this is the correct discussion regarding AMD and Windows 10 CPU support?
Note: I want this so function on my PC, Home version. Not server.
Hardware protection....

![Screenshot (20)](https://user-images.githubusercontent.com/45059948/83984868-6457ea00-a937-11ea-8471-d1c7f8176418.png)


Anyone got any suggestion for me?

![Screenshot (26)](https://user-images.githubusercontent.com/45059948/83984681-a03e7f80-a936-11ea-9d40-5bff8bba7ec8.png)

![Screenshot (23)](https://user-images.githubusercontent.com/45059948/83984884-6d48bb80-a937-11ea-91b1-7eee4513ff3e.png)

**Comment by SidShetye** [comment]
> I Have a new ASUS TUF B450-PLUS GAMING motherboard and Ryzen 5 3600Ghz CPU.
> Win 10, home, 64 bit.
> 
> BIOS says
> "HI dude, you have secure boot, lots of TPM keys and I can overclock to you hell if you want"
> "Oh yeah, CSM is on enabled, which means you can boot on both Legacy boot and UEFI! Isnt that great?
> 
> ![Screenshot (4)](https://user-images.githubusercontent.com/45059948/83984746-e1369400-a936-11ea-861b-61106db05e87.png)
> 
> Windows says:
> 
> * Nope, If you Virtualize AMD CPU, I'll CRASH
> * No secure boot loaded for you!
> * TPM looks great, please clear your TPM (after 4 times, I stopped trying)
> 
> ![Screenshot (22)](https://user-images.githubusercontent.com/45059948/83984769-f6abbe00-a936-11ea-9c11-312e5406444a.png)
> 
> * Windows further claim I have very little hardware protection. Not even ...well you check the pics.
> 
> ![Screenshot (28)](https://user-images.githubusercontent.com/45059948/83984831-3ffc0d80-a937-11ea-86b7-a44e65dc0564.png)
> 
> I Hope this is the correct discussion regarding AMD and Windows 10 CPU support?
> I want this to function on my PC, Home version. Not server.
> 
> Im interested in security. Both sides.
> 
> The hardware protection from ASUS supports Ryzen 5 fully. Which ASUS support tells me and claims "talk to Windows".
> 
> ![Screenshot (20)](https://user-images.githubusercontent.com/45059948/83984868-6457ea00-a937-11ea-8471-d1c7f8176418.png)
> 
> Anyone got any suggestion for me?
> I have done all the troubleshooters.
> 
> Is the lack of support for virtualization of AMD Ryzen CPU, the reason why my Workstation in Win 10, cant use Secureboot, EUFI, Kernal DMA, TPM device protection and GPU protection?
> 
> AURA in ASUS works pretty good though. Rainbow colors inside your computer that can sync to your keyboard colors etc haha. THAT junk works :p.
> 
> Any workaround for me?
> 
> ![Screenshot (26)](https://user-images.githubusercontent.com/45059948/83984681-a03e7f80-a936-11ea-9d40-5bff8bba7ec8.png)
> 
> ![Screenshot (23)](https://user-images.githubusercontent.com/45059948/83984884-6d48bb80-a937-11ea-91b1-7eee4513ff3e.png)
> 
> ![20200505_202246](https://user-images.githubusercontent.com/45059948/83985302-eb599200-a938-11ea-8852-ceab39eb6eb9.jpg)
> ![20200505_202300](https://user-images.githubusercontent.com/45059948/83985304-eeed1900-a938-11ea-9675-cd22773e7c2a.jpg)
> ![20200505_202620](https://user-images.githubusercontent.com/45059948/83985308-f3193680-a938-11ea-93e5-fb4520a6657c.jpg)
> 
> I'm going nuts not being able to use the hardware protection that's built in the system!
> 
> Even Vaulttec-Boy has any idea what to do.
> 
> ![Jonah_PC_Vaultguy1](https://user-images.githubusercontent.com/45059948/83985641-42ac3200-a93a-11ea-91a7-d02f85e59553.jpg)

I wish you had spent 1/10th the time of your post on researching the issue you’re replying to.

**Comment by JouMxyzptlk** [comment]
> I Have a new ASUS TUF B450-PLUS GAMING motherboard and Ryzen 5 3600Ghz CPU.
> Win 10, home, 64 bit.
> "HI dude, you have secure boot, lots of TPM keys and I can overclock to you hell if you want"

You are totally invading with something so off topic it can only be described as Trump-Whataboutism. Please delete your post.
I see "Driver Magician", so I suspect quite a few other things are wrong with your Windows as well. Windows is, despite its reputation, a quite open System and does not prevent any wild program from the internet f*-ing it up.
Now again: Please leave this thread, delete you post, and ask somewhere were TPM + ASUS is the topic, preferably an ASUS forum.

**Comment by plukawski** [comment]
@Kurnn please stop doing off topic stuff...

**Comment by ssoor** [comment]
I'm very happy to see the news that Windows 10 can do it, and look forward to when Server 2019 will be supported.

**Comment by plukawski** [comment]
@ssoor From what I see MS does not want to tell us about that feature (there is literally no word from MS on any official channel I know). So to check that on Windows Server you will have to monitor the insider builds by your own I think. 
The case of not informing community of introducing such a big feature is very strange...

**Comment by lvasiliou** [comment]
@plukawski i believe that is because hyper-v is intended first for Windows Server. Noticed it even before Windows 10.

**Comment by lvasiliou** [comment]
OK a symptom of the insider build is that I cant install the WSL2 kernel update, it says i must have WSL2 installed. And because of that WSL2 does not work either which also breaks Docker. A warning for those who bump to this build without WSL2 fully configured.

**Comment by lvasiliou** [comment]
@Kurnn update your UEFI from Asus, disable CSM Compatibility, make sure your TPM is enabled and you cleared it from UEFI (if you have one), ensure secure boot is set to Windows OS. I am not sure if AMD is bundling a TPM with the Ryzen 3/5/7/9 as they do with the Threadripper, you might need to buy one or switch to Firmware TPM in the UEFI
Also, this is the wrong forum

**Comment by plukawski** [comment]
@lvasiliou I do not agree about hyper-v being related specially to Windows Server. When the nested virtualization was introduced in hyper-v in year 2015 in the insider build for Intel it was officially announced, see here: https://docs.microsoft.com/en-us/archive/blogs/virtual_pc_guy/nested-virtualization-in-windows-10-build-10565 

So this is very strange we do not have any official information about that topic now.

**Comment by Larswa** [comment]
Hey @lvasiliou     My docker was broken, out of the gate, after installing the fast ring build, but a docker factory reset did the trick for me. Have not been attempting a kernel upgrade for WSL2. 

Oh .. and it defaults to the WSL2 instance I have running.

**Comment by lvasiliou** [comment]
@Larswa i sorted it by extracting the kernel update msi into windows\system32\lssxs\tools. I never had wsl2 in the first place so different scenario

**Comment by Larswa** [comment]
@lvasiliou  Niiiice work man!  :-)

**Comment by wbsmolen** [comment]
Hey everyone - there will be a [blog post](https://techcommunity.microsoft.com/t5/virtualization/bg-p/Virtualization) on this capability in the near future. Keep an eye on the blog this week!

**Comment by plukawski** [comment]
Can someone tell me what family of AMD cpus are supported for that capability? Is only Zen2 supported or Zen1 and Bulldozer as well ?

**Comment by neico** [comment]
Pretty sure this applies to all CPU's that support AMD-V (SVM) and Nested Paging (NPT), in contrast to Intel's VT-x and EPT

see:
https://en.wikipedia.org/wiki/X86_virtualization#AMD_virtualization_(AMD-V)
https://en.wikipedia.org/wiki/Second_Level_Address_Translation
https://docs.microsoft.com/en-us/virtualization/hyper-v-on-windows/user-guide/nested-virtualization#prerequisites

**Comment by plukawski** [comment]
@neico I don't know how the information from your links is confirming your statement. For now we have zero official information from MS about that feature and its limitations.
However I have just tested on my laptop with Ryzen 5 2500U CPU (Zen1) and can confirm that nesting is working on it just fine. So I can tell from this experience that at least Zen1 is supported, however I do not have any older AMD cpu currently to check if nesting is working on them as well.

**Comment by JouMxyzptlk** [comment]
> Can someone tell me what families of AMD cpus are supported for that capability? Is only Zen2 supported or Zen1 and Bulldozer as well ?

If you have a pre-Zen CPU: Make a backup from which you can boot and restore. Then just try! I did create a "disaster recovery capable" backup, in my case with drivesnapshot which works when booted from a windows stick. There are other tools out there which can do the same, some of the are even free.
If making such a backup is beyond your scope learn / train / try how to do it, and then update to that insider build to get your answer.
You can obtain that insider build from uupdump as well https://uupdump.ml/known.php - but you still should backup before trying.

**Comment by mycail** [comment]
Nested virtualization's working for me on latest Insider (fast ring) build. Version 2004, Build 19640.1. AMD Ryzen 3950X CPU. Hyper-V host, Ubuntu guest, nested Ubuntu guest. Weary about running an Insider build as I've gotten some GSODs before and had to reinstall completely... but glad to see this feature has been implemented. Hope to see this in the stable/non-Insider build soon!

**Comment by plukawski** [comment]
@JouMxyzptlk currently I do not have access to pre Zen AMD CPU. However for testing I prefer using VHD Native Boot feature instead of backing up the entire system and install Insider Build as a main system. I would also suggest you to try it - it greatly eases the coexistence of multiple Windows OS on a single machine. I use a tool called **WinNTSetup3** for creating such VHDs directly from the ISO. When you look into my github repository you will find that I have also created a console tool for moving your main OS to the VHD Native Boot :)

**Comment by Paranoid-OS** [comment]
> Update on tests with 19640 nested virtualization: All 64 Bit Windows OSes work fine. The only Windows 32 bit OS that works as nested guest is Windows NT 4.0 (you read right!). Windows 2000/Win XP/Server 2003/Vista 32 bit/2008 32 bit/win7 32 bit all come up with a blue screen. They work on the real Win10 host though, exported and imported into the virtual server 2019 nested-enabled guest -> fail with blue screen.

So are you saying it works without having to create the VM in powershell in Windows build 19640?
Or that it works when creating the VM in powershell?

> Confirmed! Update-VMVersion does not work yet, and you have to create the machine with Powershell using:
> New-VM -Name "S2019 Nested Test" -Version 9.3 -Generation 2
> If you don't supply a version a 9.0 machine is created. Currently installing Server 2019 to run part two of the test, activating Hyper-V in there.

And would it work in build 20150? I have build 19619, so I think I have to upgrade, and MS is offering me 20150 now

**Comment by JouMxyzptlk** [comment]
> So are you saying it works without having to create the VM in powershell in Windows build 19640?
> Or that it works when creating the VM in powershell?

Why should I create the nested, aka 2nd level, guest using powershell?

**Comment by Paranoid-OS** [comment]
my poor choice of words. I meant creating a machine in hyper-v, not nested
The post above suggested I had to create the first machine using powershell, or it wouldn't work

**Comment by pantasio** [comment]
> I Have a new ASUS TUF B450-PLUS GAMING motherboard and Ryzen 5 3600Ghz CPU.
> Win 10, home, 64 bit.
> 
> BIOS says
> "HI dude, you have secure boot, lots of TPM keys and I can overclock to you hell if you want"
> "Oh yeah, CSM is on enabled, which means you can boot on both Legacy boot and UEFI! Isnt that great?
> 
> ![Screenshot (4)](https://user-images.githubusercontent.com/45059948/83984746-e1369400-a936-11ea-861b-61106db05e87.png)
> 
> Windows says:
> 
> * Nope, If you Virtualize AMD CPU, I'll CRASH
> * No secure boot loaded for you!
> * TPM looks great, please clear your TPM (after 4 times, I stopped trying)
> 
> ![Screenshot (22)](https://user-images.githubusercontent.com/45059948/83984769-f6abbe00-a936-11ea-9c11-312e5406444a.png)
> 
> * Windows further claim I have very little hardware protection. Not even ...well you check the pics.
> 
> ![Screenshot (28)](https://user-images.githubusercontent.com/45059948/83984831-3ffc0d80-a937-11ea-86b7-a44e65dc0564.png)
> 
> I Hope this is the correct discussion regarding AMD and Windows 10 CPU support?
> I want this to function on my PC, Home version. Not server.
> 
> Im interested in security. Both sides.
> 
> The hardware protection from ASUS supports Ryzen 5 fully. Which ASUS support tells me and claims "talk to Windows".
> 
> ![Screenshot (20)](https://user-images.githubusercontent.com/45059948/83984868-6457ea00-a937-11ea-8471-d1c7f8176418.png)
> 
> Anyone got any suggestion for me?
> I have done all the troubleshooters.
> 
> Is the lack of support for virtualization of AMD Ryzen CPU, the reason why my Workstation in Win 10, cant use Secureboot, EUFI, Kernal DMA, TPM device protection and GPU protection?
> 
> AURA in ASUS works pretty good though. Rainbow colors inside your computer that can sync to your keyboard colors etc haha. THAT junk works :p.
> 
> Any workaround for me?
> 
> ![Screenshot (26)](https://user-images.githubusercontent.com/45059948/83984681-a03e7f80-a936-11ea-9d40-5bff8bba7ec8.png)
> 
> ![Screenshot (23)](https://user-images.githubusercontent.com/45059948/83984884-6d48bb80-a937-11ea-91b1-7eee4513ff3e.png)
> 
> ![20200505_202246](https://user-images.githubusercontent.com/45059948/83985302-eb599200-a938-11ea-8852-ceab39eb6eb9.jpg)
> ![20200505_202300](https://user-images.githubusercontent.com/45059948/83985304-eeed1900-a938-11ea-9675-cd22773e7c2a.jpg)
> ![20200505_202620](https://user-images.githubusercontent.com/45059948/83985308-f3193680-a938-11ea-93e5-fb4520a6657c.jpg)
> 
> I'm going nuts not being able to use the hardware protection that's built in the system!
> 
> Even Vaulttec-Boy has any idea what to do.
> 
> ![Jonah_PC_Vaultguy1](https://user-images.githubusercontent.com/45059948/83985641-42ac3200-a93a-11ea-91a7-d02f85e59553.jpg)

You must upgrade win 10 pro or win 10 enterprise to run hyper-v

**Comment by Paranoid-OS** [comment]
> You must upgrade win 10 pro or win 10 enterprise to run hyper-v

and also win 10 edu and server

And could you please delete all those photos? This is an issue about nested virtualization on AMD CPUs, not about your PC and the incompatibility of some Windows versions with hyper-v (this is **not** nested virtualizaion!)

**Comment by JouMxyzptlk** [comment]
When following the build numbers and which will be the future 20H2 I make following prediction: Nested virtualization will not be included in 20H2, which will be 19042.*, a rather small step like 1903-1909 step probably thanks to COVID19.
Insider build 19645.* was from 10th June and maybe that was planned for 21H1 with official support for nested on Ryzen. Currently newest insider build is 20161.* for AMD and 20170.* for Intel due to some mentioned bug on AMD CPUs with 20170.*, looks like skipping 196* builds - maybe due to COVID19 as well, not wasting resources on that branch.

**Comment by lvasiliou** [comment]
> 
> 
> @pantasio
> 
> > You must upgrade win 10 pro or win 10 enterprise to run hyper-v
> 
> and also win 10 edu and server
> 
> And could you please delete all those photos? This is an issue about nested virtualization on AMD CPUs, not about your PC and the incompatibility of some Windows versions with hyper-v (which is **not** about nested virtualization!)

And dont forget to clean your desk and do cable management 🤣

**Comment by ImreKovats** [comment]
You guys! Quick question: Are Vishera core CPUs ever gonna be supported?

**Comment by lvasiliou** [comment]
So I am on 19042.423 and I cannot select option 9.3 anymore:The VM version 9.3 of the virtual machine 'New Virtual Machine' is not supported.

**Comment by JouMxyzptlk** [comment]
> So I am on 19042.423 and I cannot select option 9.3 anymore:The VM version 9.3 of the virtual machine 'New Virtual Machine' is not supported.

It will never work with 19**0**??.??? version. Minimum build requires is 19**6**40 (or the current 20180)

**Comment by lvasiliou** [comment]
> 
> 
> > So I am on 19042.423 and I cannot select option 9.3 anymore:The VM version 9.3 of the virtual machine 'New Virtual Machine' is not supported.
> 
> It will never work with 19**0**??.??? version. Minimum build required is 19**6**40 (or the current 20180).

OK which channel is that? I am getting totally confused with the Windows Insider programme and which channel gives you what. Probably easier to run linux

**Comment by Paranoid-OS** [comment]
> OK which channel is that? I am getting totally confused with the Windows Insider programme and which channel gives you what. Probably easier to run linux

DEV channel is the bleed'n edge, so DEV should work(it was build 20180 last time I checked) if the other ones aren't higher than 19640

**Comment by JouMxyzptlk** [comment]
Nice news: Windows Server Insider LTSC is now build 20201 - which means nested virtualization should work with real server OS as well and not just Windows 10.
https://blogs.windows.com/windows-insider/2020/08/26/announcing-windows-server-vnext-preview-build-20201/
But my server is my old i7-4960x and no Threadripper/Epyc, My secondary server with 192 GB RAM is even older with Sandy Bridge CPUs and only powered up for specific tasks which need such an amount of RAM, such as encoding 16k videos.
Anybody got me a Threadripper/Epyc System with 192 GB RAM? ;)

**Comment by JouMxyzptlk** [comment]
Nice news: Windows Server Insider LTSC is now build 20201 - which means nested virtualization should work with real server OS as well and not just Windows 10.
https://blogs.windows.com/windows-insider/2020/08/26/announcing-windows-server-vnext-preview-build-20201/
But my server is my old i7-4960x and no Threadripper/Epyc, My secondary server with 192 GB RAM is even older with Sandy Bridge CPUs and only powered up for specific tasks which need such an amount of RAM, such as encoding 16k videos.
Anybody got me a Threadripper/Epyc System with 192 GB RAM? ;)
As for now I will have to try nested in a virtual Server-Next within a virtual Server-Next within a virtual Server-Next within the real Windows 10 20201...

**Comment by JouMxyzptlk** [comment]
Nice news: Windows Server Insider LTSC is now build 20201 - which means nested virtualization should work with real server OS as well and not just Windows 10.
https://blogs.windows.com/windows-insider/2020/08/26/announcing-windows-server-vnext-preview-build-20201/
But my server is my old i7-4960x and no Threadripper/Epyc, My secondary server with 192 GB RAM is even older with Sandy Bridge CPUs and only powered up for specific tasks which need such an amount of RAM, such as encoding 16k videos.
Anybody got me a Threadripper/Epyc System with 192 GB RAM? ;)
As for now I will have to try nested in a virtual Server-Next within a virtual Server-Next within a virtual Server-Next within the real Windows 10 20201...

**Comment by JouMxyzptlk** [comment]
For those who can: Test Windows Server 20201.1000 Insider. I've already reported a few weirdness when trying to run it on Ryzen 3900x as nested 2nd-level guest (Host Win10 20201.1000, first level Server 20201.1000, second level 20201.1000 (version 9.3) normal reboot fail, second level Server 2019 (version 9.3 machine) activate hyper-v bricks the virtual machine, even though it should work.
They need the feedback before they release the 21h1 Server (and Windows 10).
https://blogs.windows.com/windows-insider/2020/08/26/announcing-windows-server-vnext-preview-build-20201/

**Comment by lygstate** [comment]
does kvm + win10+hyperv+ wsl2 works now？

**Comment by lvasiliou** [comment]
Not sure what the situation is with AMD nested virtualization on KVM but that chain works fine on hyper-v

**Comment by lvasiliou** [comment]
Probably not the test way to request this but can we get a dedicated ui app for hyper-v? Experiencing quite a few issues free past few years with the archaic mmc like having to connect to the local hyper-v service every startup or having no VMs showing up and having to reregister them only to be told they are registered on the server already. Also my default switch seems to be broken past 2 builds.

**Comment by JouMxyzptlk** [comment]
Honestly: Why do people post so much OT stuff here?
KVM + Hyper-V: You are in the wrong thread!
Hyper-V GUI: Use Powershell. The Management-Service style is standard among many OS-es. Do you want a web GUI, enforcing you to run and maintain IIS on a Hyper-V host? Also: Wrong place to complain, make your own suggestion.
HV-Switch broken complaint: Oh come one, you are on Insider-Dev, I recently had t restore the pre-insider backup since the appx-upgrade broke with 20201 and all following builds consequently failed to upgrade 1% before the finishing line. But why don't you use the feedback Hub and complain here?

**Comment by Dresel** [comment]
> KVM + Hyper-V: You are in the wrong thread!

@JouMxyzptlk Any hints about the "right" thread? This is the one you will findwhen searching about "kvm nested virtualization windows ryzen"...

**Comment by lygstate** [comment]
> Honestly: Why do people post so much OT stuff here?
> KVM + Hyper-V: You are in the wrong thread
Also confusing!
> Hyper-V GUI: Use Powershell. The Management-Service style is standard among many OS-es. Do you want a web GUI, enforcing you to run and maintain IIS on a Hyper-V host? Also: Wrong place to complain, make your own suggestion.
> HV-Switch broken complaint: Oh come one, you are on Insider-Dev, I recently had t restore the pre-insider backup since the appx-upgrade broke with 20201 and all following builds consequently failed to upgrade 1% before the finishing line. But why don't you use the feedback Hub and complain here?

**Comment by Paranoid-OS** [comment]
If you want to use remoteFX with this, you can just create the VM as a gen 1, nested works with gen 1 as well

**Comment by joint007** [comment]
> > Honestly: Why do people post so much OT stuff here?
> > KVM + Hyper-V: You are in the wrong thread
> 
> Also confusing!

I need to ask, why?
Caption of this thread is:"AMD nested virtualization?"
It is in "MicrosoftDocs/Virtualization-Documentation"

Nested virtualization on Windows and AMD is broken everywhere, not only in Hyper-V on first-level host.
Most of people want run virtual-machines with Windows on KVM, and have choice to run second-level virtual-machines in Windows.

My story is simple, i have two Windows Servers in KVM, one of them on Intel, one on Ryzen.
Everything works and run very well, but if i try to run second-level virtual-machine on AMD, it always get stuck. 
But on Intel, no problem.

Its is only problem with Windows and Microsoft, who threw AMD overboard and did not implement this functionality.

Where is the right place to give notice about the problem, which by all indications also belongs here?

**Comment by JouMxyzptlk** [comment]
As for Insider Build 21313.1000, the first one that fixed my PREOOBE Problem and installed fine since June 2020: The configuration-version can now be 10.0. New created VMs using the GUI will have version 10.0 automatically. You can upgrade an existing machine to Version 10.0 in the GUI. Now you ONLY need to run Set-VMProcessor -ExposeVirtualizationExtensions $true -VMName <your vm name>.

**Comment by JouMxyzptlk** [comment]
The same for Server 2022 newest preview, Build 20303.1: As written above for Windows 10 21313.
Download at https://www.microsoft.com/en-us/software-download/windowsinsiderpreviewserver

**Comment by 0xF5F5F5** [comment]
I'm running Windows on a VM on Linux (libvirt, kvm, qemu).
Still have issues running Gen 2 VMs. Gen 1 VMs are able to run, Gen 2 VM shows only a black screen at all times and I notice one CPU core on host is maxed out in kernel-mode usage. I believe this is why Windows Sandbox shows a generic error when I start it up. (I don't remember how WSL2 fails to start but I do know that it fails to start)

(I'm not sure if the Gen 1 VM is running with AMD virtualization extensions, so far I've only booted a small Linux ISO. I can try to run "openssl speed" next time.

**Comment by Joachim-Otahal** [comment]
Oh great, announcing Windows 10 21H2 as insider build number 19044.1147 - I hate being toyed with. How long will Microsoft keep on mocking us? Do the want to reserve that feature for Windows 11? The code is there since May 2019, and yet they only still offer the upcoming Server 2022 as the only "official" way, or having to use an insider build (which nukes my TV card)... This is SO annoying.

**Comment by joergbattermann** [comment]
Given that Windows Server 2022 is officially out as of last week, has anyone had the chance to try out if it comes with nested virtualization on AMD (Epyc) platforms?

**Comment by plukawski** [comment]
@joergbattermann  Yes - it fully supports nesting on AMD Hardware, its default HYPER-V configuration version is 10.0, and since 9.3 version nesting is fully supported on AMD platform. I have even downloaded an evaluation ISO and checked by myself and I can confirm, that nesting is working fine on my Ryzen 7 3700X.

However, I don't know if this is related only to AMD (I don't have Intel system now to test this) or is it just nesting itself, but as soon as I enable nesting for a VM, I see very significant performance degradation in that VM even when the VM is currently not running any nested virtual machine inside. For example Cinebench R20 score in that VM before enabling nested virtualization is 3100-3400, but after enabling nesting the score goes down to 1600-2100 levels (no nested virtual machine is running in background). Such degradation is noticeable everywhere - even opening explorer window start to be sluggish. As soon as I disable nesting for the VM its performance comes back to normal levels. Does anyone is also observing such issues?

**Comment by Joachim-Otahal** [comment]
You see correct. The performance of nested virtualization even improved with Windows 11 / S2022, but it is still true:
https://virtualizationreview.com/blogs/everyday-virtualization/2015/07/testing-windows--server-2016-with-nested-hyper-v.aspx

**Comment by plukawski** [comment]
@Joachim-Otahal I have seen that article before, but I think it is talking about performance hit seen in the Level 2 Guest (that is the nested virtual machine) not the virtual machine that hosts it. However I may be wrong. 
A year ago I was working on nested virtualization done using VMWare ESXi on the same hardware and didn't notice such performance degradations, however I do not have the exact numbers now. Maybe I will try to check if VMWare ESXi is introducing the same level of performance degradation for level 1 guests...

**Comment by plukawski** [comment]
@Joachim-Otahal I have recently tested if there is any performance degradation after enabling nesting to a VM on the Intel platform (i7-10700). It turned out that there was no meaningful performance degradation there at all - the VM after exposing virtualization extensions to it and installing hyper-v role did perform exactly the same as before doing that. 
On AMD system the performance hit in such situation is in range 10-40% (the performance is highly unstable between benchmark runs). This poses a question - why is this happening? Having so significant performance drop when nesting is enabled can render this feature totally useless on AMD platform.
On AMD platform I tested that on both Windows 11 and Windows Server 2022 installed on the host. On Intel platform I tested that having Windows 10 20H2 on the host.
In VM I used Windows 11 most recent build (.194) for both Intel and AMD.

**Comment by Joachim-Otahal** [comment]
> @Joachim-Otahal I have recently tested if there is any performance degradation after enabling nesting to a VM on the Intel platform (i7-10700). It turned out that there was no meaningful performance degradation there at all - the VM after exposing virtualization extensions to it and installing hyper-v role did perform exactly the same as before doing that.
> On AMD system the performance hit in such situation is in range 10-40% (the performance is highly unstable between benchmark runs). This poses a question - why is this happening? Having so significant performance drop when nesting is enabled can render this feature totally useless on AMD platform.
> On AMD platform I tested that on both Windows 11 and Windows Server 2022 installed on the host (Ryzen 3700X). On Intel platform I tested that having Windows 10 20H2 on the host.
> In VM I used Windows 11 most recent build (.194) for both Intel and AMD.

I cannot tell. I'm not associated by Microsoft. There are too many variables from energy saving settings in Windows, core parking, BIOS/UEFI update, CPU Microcode (which comes with the BIOS) and so on. To be honest: I won't invest time checking for that specific issue on my i7-4960x in comparison to my Ryzen 5950x. I won't be getting a newer Intel system in the near future, rather phasing out my i7-4960x which replaced my older server-system sind I got the Ryzen 2700x.
You might contact your hardware manufacturers and Microsoft directly for better answers.

**Comment by plukawski** [comment]
@Joachim-Otahal I don't want you to invest any time in this :) I just wanted to share my own findings. However, if you find it curious, I suggest to check this on your Ryzen machine if you are experiencing the same degradation? If the problem does not reproduce for you then something is clearly just wrong with my hardware configuration. 
It is easy to reproduce, here are the steps:
1. Create a new VM with at least 8 vcores (without enabling nesting for now) and install Windows 11 preview inside
2. After all updates are installed and system is idle Install CPU-z in that VM, run built in benchmark 3 times and write the avg values.
3. Shutdown VM, enable nesting for that VM and then install HYPER-V role inside it.
4. Rerun benchmark from CPU-Z in that VM and compare the results from step 2.
I found this to be the fastest way to check if there are any performance regressions. For me the regression is like already mentioned between 10% and 40%, which is huge and in my opinion unacceptable.

**Comment by 0xF5F5F5** [comment]
> 
> 
> I'm running Windows on a VM on Linux (libvirt, kvm, qemu).
> Still have issues running Gen 2 VMs. Gen 1 VMs are able to run, Gen 2 VM shows only a black screen at all times and I notice one CPU core on host is maxed out in kernel-mode usage. I believe this is why Windows Sandbox shows a generic error when I start it up. (I don't remember how WSL2 fails to start but I do know that it fails to start)
> 
> (I'm not sure if the Gen 1 VM is running with AMD virtualization extensions, so far I've only booted a small Linux ISO. I can try to run "openssl speed" next time.)

Soon after I made this comment I tested and it was clear that the gen1 VM was performing as expected.
But that's about it. I'm on Windows 11 (21H2, 22000.194) and gen2 VMs still do not start. I see a 'vmmem' process use 100% of a single cpu thread whenever I start a gen2 VM. I just tried Windows Sandbox and the same thing happened.

**Comment by Daasin** [comment]
Last I was told, this was working in the insider builds for Windows 10. 
 
Has it been taken out? I ask because not all Ryzen users are able to officially install Win 11 :/

**Comment by Daasin** [comment]
Last I was told, this was working in the insider builds for Windows 10. 
 
Has it been taken out? I ask because not all Ryzen users are able to officially install Win 11 :/  

Especially given that it was explicitly said to be working on 1st Gen Ryzen @SidShetye @johncslack

**Comment by Joachim-Otahal** [comment]
> Last I was told, this was working in the insider builds for Windows 10.
> 
> Has it been taken out? I ask because not all Ryzen users are able to officially install Win 11 :/

Yes, it has been taken out. VERY annoying. This is the major reason what I switched to Windows 11 so early on. The only other thing I find actually positive are robocopy /iorate options. As for the rest solala, a messed up taskbar, and a new useless overlay over the old explorer context menu (which you can deactivate with a registry key), explorer renaming and generally slower explorer and so on. If you don't actually need nested virtualization, wait for the next service pack. The other optimizations below the messed up UI don't compensate unless you have actual use for them. I use nested-v for deduplication on the first level guest (server 2019, soon server 2022), holding about 15 test guest machines with a deduplication rate between 60% and 70%, saving 400 GB of SSD space.
Nobody knows if or when they might give that feature to Windows 10. I suspect, currently, never.
If you want a more useful version of it use Server 2022, but that has some other side effects when it comes to gaming, for example.

**Comment by Daasin** [comment]
Shame, I would've used a non-standard way of installing Win 11 for first gen nest-virt but blocked updates and compatibility are a hassle and I expect the amount of hassles to grow for non-officially supported CPUs 

It's a small feature that they already proved can be supported and then it did work while we tested it but they took it out. Almost a forced upgrade when 7th Gen Intel was sitting pretty.

**Comment by Joachim-Otahal** [comment]
If you need nested-v with a less messed up UI use Server 2022. You can convert the 180 days eval version to an unlimited-forever-eval version. Brings some restrictions regarding UI customization, which you can work around too. As fas as gaming is concerned: I never tested it, it should be between windows 10 and windows 11 performance-wise since the kernel is newer. Good luck!

**Comment by 0xF5F5F5** [comment]
> Yes, it has been taken out. VERY annoying. 

lol, wow. I installed the Hyper-V role again after reading your comment and I got a BSOD on reboot (HYPERVISOR_ERROR I think). Later I rebooted into safe mode and Windows automatically reverted the installation of Hyper-V (it knew it failed to boot multiple times). So I guess this has to do with what you say. I can't believe it seems like Windows 11 regressed on support for this.

Either way, I was never able to get generation-2 VMs working (on Windows VM on Linux KVM, on Ryzen 7 3700X), but now it can't even start Hyper-V. When Hyper-V used to start up on Windows 10, the generation 2 VMs would behave like others have described, they are just unresponsive and there is high activity on one cpu.

btw, is Windows 11 missing the bootloader option for hypervisor startup? Before trying to reboot into safe mode, I tried running bcdedit in recovery environment, but the hypervisor parameter was not listed like I expected, and this is the first time I looked at bcdedit in Windows 11, so I guess it is a new thing. Meaning there may be no easy way to disable hypervisor with BCD.

> a messed up taskbar
lol yep, I've been using ExplorerPatcher lately, it's a brilliant tool and it's on GitHub 👍 I would be happy to use the Windows 11 taskbar if you could adjust its height and if they didn't deprecate toolbars. I'm not surprised toolbars don't work, the new toolbar must be a lot of new code, but it kinda sucks that toolbars will probably remain unsupported.

**Comment by Joachim-Otahal** [comment]
> I installed the Hyper-V role again after reading your comment and I got a BSOD on reboot (HYPERVISOR_ERROR I think).
> Either way, I was never able to get generation-2 VMs working (on Windows VM on Linux KVM, on Ryzen 7 3700X),

You use Linux KVM as main hypervisor? Your issues are COMPLETELY different from here, and at least 90% off-topic. Your problem ist first in the KVM Level, and then, maybe, the Windows OS. If your KVM does it right you can activate Hyper-V in your virtual machine running Windows 8(.1) or Server 2008 (R2) or later. If that does not work it is a KVM problem, not a Windows problem. You should contact your Linux groups first. Maybe a Linux update changed something in your KVM, and now it doesn't do what it could a few Month before?

**Comment by Szeraax** [comment]
This [Dell article](https://www.dell.com/support/kbdoc/en-au/000132088/third-party-nested-virtualization-in-windows-server-2019) explains my goal nicely where I have bare-metal hyper-V and then I install ProxMox, ESXi, or other linux hypervisors to run VMs and Containers in. It has a pretty picture of what I want to do right at the top, even! :)

And its requirements section rightly states that you need: 

> An Intel processor with VT-x and EPT technology - nesting is currently supported on systems with Intel processors only.

I'm left here wondering why any old Intel CPU can do this, but all Ryzen processors are getting left out in the cold. Do I spend more money for an Intel CPU that I don't like? My whole household is AMD at present.

**Comment by plukawski** [comment]
@Szeraax , please note that the article from Dell site was last updated before the Windows Server 2022 launched. At that time there was no support for nesting on AMD processors in HYPER-V (it was only in the Insider builds). However The Windows Server 2022 now has official support for nesting on AMD processors, so in my opinion you can try to run it.

**Comment by Szeraax** [comment]
@plukawski 

My understanding is that this is ONLY for Hyper-V in Hyper-V. Not KVM/Linux in Hyper-V.

**Comment by Joachim-Otahal** [comment]
@Szeraax Since Windows Build 19640 (by that time Windows 10 insider fast ring), which was about May/June 2020, nested virtualization works with AMD, proven. It seems that any Windows build, doesn't matter whether Windows 10, 11 or Server 2022, with a build number 19640 or higher can do it. Unfortunately Microsoft in its unknown wisdom decided to keep that feature from Windows 10 since all official upgrades since 20H1 are built 19041/19042/19043/19044, and I SUSPECT the upcoming 22H1 Windows 10 build will be 19045. Therefore deliberately forcing to upgrade to Server 2022 or Windows 11 if you want nested-V on AMD. You can download the trial version of Server 2022, convert it to an "endless trial" instead of 180 days trial (google on how to), and play with that one at home. Don't do this on company machines though.

**Comment by Joachim-Otahal** [comment]
> @plukawski
> 
> My understanding is that this is ONLY for Hyper-V in Hyper-V. Not KVM/Linux in Hyper-V.

You are wrong here: The first level of hypervisor must support it on the CPU architecture, no matter whether you use Windows/VMware/Linux/etc, So it does apply to "KVM as guest inside a Hyper-V VM". And the other way around too. Though Microsoft was the last from all popular hypervisors to support it, all others were there earlier.

**Comment by Szeraax** [comment]
I'll have to go spin up a new 2022 server and see. I bet you're right that its like not backporting DNS cmdlets for windows 8 to windows 7 despite it working fine if you import the module on a Win7 machine. 

 There is [this MS doc](https://docs.microsoft.com/en-us/system-center/vmm/vm-nested-virtualization?view=sc-vmm-2022) that at least discourages it: 

> Virtualization applications other than Hyper-V are not supported in Hyper-V virtual machines, and are likely to fail. This includes any software that requires hardware virtualization extensions.

**Comment by Daasin** [comment]
> @Szeraax , please note that the article from Dell site was last updated before the Windows Server 2022 launched. At that time there was no support for nesting on AMD processors in HYPER-V (it was only in the Insider builds). However The Windows Server 2022 now has official support for nesting on AMD processors, so in my opinion you can try to run it.

Including Ryzen 1000? It's a shame since it worked for I sider users that had it and then changed mind last minute after giving their word :/ Feels misleading

**Comment by Szeraax** [comment]
If this is considered done, any reason for this topic to stay open? I always get a good feeling when people close with WONTFIX.

**Comment by Daasin** [comment]
> please note that the article from Dell site was last updated before the Windows Server 2022 launched. At that time there was no support for nesting on AMD processors in HYPER-V (it was only in the Insider builds). However The Windows Server 2022 now has official support for nesting on AMD processors, so in my opinion you can try to run it.  
 @plukawski  

Including Ryzen 1000? It's a shame since it worked for I sider users that had it and then changed mind last minute after giving their word :/ Feels misleading

**Comment by rifaterdemsahin** [comment]
> s, but all Ryzen processors are getting left out in the cold. Do I spend more money for an Intel CPU that I don't like? My whole household is AMD at present.

did you find a way on the amd ?

**Comment by Daasin** [comment]
@brasmith-ms Would there be updates on this by any chance or should we just close the issue :/ 

Could've sworn that the feature achieved full functionality on W10 but then was dropped entirely :(

**Comment by Joachim-Otahal** [comment]
It is solved. Windows 11 and Server 2022 can do nested-v with AMD. The only reason why I am running Windows 11. Microsoft tactical decision.
Please close this issue.

**Comment by SidShetye** [comment]
Personally, I've moved on and the use case is no longer applicable for me. @Joachim-Otahal - can you point to any docs/artifacts to confirm nested Hyper-V virtualization works on AMD processors? For the sake of other AMD users facing this issue, I'd like to keep this open till we can confirm.

**Comment by Joachim-Otahal** [comment]
> @Joachim-Otahal - can you point to any docs/artifacts to confirm nested Hyper-V virtualization works on AMD processors?

Microsoft should pay me for this. And you too, IMHO :D!
HowTo: https://docs.microsoft.com/en-us/virtualization/hyper-v-on-windows/user-guide/nested-virtualization
News on Server 2022 and Windows 11 regarding this: https://docs.microsoft.com/en-us/windows-server/get-started/whats-new-in-windows-server-2022

**Comment by SidShetye** [comment]
Thanks. Confirmed and closing.

From the first link

> Run Hyper-V in a Virtual Machine with Nested Virtualization
> ...
> Prerequisites
> ...
> AMD EPYC/Ryzen processor or later 
> - The Hyper-V host must be Windows Server 2022/Windows 11 or greater
> - VM configuration version 10.0 or greater

From the 2nd link

> Nested virtualization is a feature that allows you to run Hyper-V inside of a Hyper-V virtual machine (VM). Windows Server 2022 brings support for nested virtualization using AMD processors, giving more choices of hardware for your environments. More information can be found at the [nested virtualization documentation](https://docs.microsoft.com/en-us/virtualization/hyper-v-on-windows/user-guide/nested-virtualization).

**Comment by valinet** [comment]
Isn't there any way to force it to work on LTSC 2021 as well, or at least backport it there officially?
